lib/zkProving.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  MembershipProver,
+  Tree,
+  Poseidon,
+  defaultPubkeyMembershipPConfig,
+} from "@personaelabs/spartan-ecdsa";
+import { loadSigmojis, updateSigmoji } from "@/lib/localStorage";
+import { cardPubKeys } from "./cardPubKeys";
+import { importPublic, ecrecover } from "@ethereumjs/util";
+import { sha256 } from "js-sha256";
+import { serializeSigmojiZKP } from "@/lib/types";
+import { parseDERSignature, findV } from "@/lib/signatureUtils";
+
+export interface ProverWasm {
+  poseidon: Poseidon;
+  prover: MembershipProver;
+}
+
+export async function initWasm(): Promise<ProverWasm> {
+  const poseidon = new Poseidon();
+  await poseidon.initWasm();
+
+  const prover = new MembershipProver({
+    ...defaultPubkeyMembershipPConfig,
+    enableProfiler: true,
+  });
+
+  await prover.initWasm();
+
+  return { poseidon, prover };
+}
+
+export const makeProofs = async (proverWasm: ProverWasm) => {
+  const treeDepth = 20;
+  const pubKeyTree = new Tree(treeDepth, proverWasm.poseidon);
+
+  for (const entry of Object.entries(cardPubKeys)) {
+    const secondaryPublicKeyRawUint8Array = new Uint8Array(
+      (entry[1].secondaryPublicKeyRaw.match(/.{1,2}/g) || []).map((byte) =>
+        parseInt(byte, 16)
```

Comment:
Would be useful to refactor this into utils, perhaps named `publicKeyHexToUint8Array`

Reply:
Very good point, cleaned up code a lot!